server/application/rest/service_commitment.py

```diff
@@ -0,0 +1,66 @@
+"""
+Module for handling service commitment API endpoints.
+Provides endpoints for creating and retrieving service commitments.
+"""
+from flask import Blueprint, request, jsonify
+from application.rest.work_shift import get_user_from_token
+from use_cases.add_service_commitments import add_service_commitments, get_service_commitments
+from repository.mongodb.service_commitments import MongoRepoCommitments  # Adjust the import path as needed
+
+blueprint = Blueprint("service_commitment", __name__)
+
+# Create a MongoRepoCommitments object to connect to the database.
+repo = MongoRepoCommitments()
+
+@blueprint.route("/service_commitment", methods=["POST"])
+def create_service_commitment():
+    """
+    Handle POST request to create service commitments.
+    Extract user info from Authorization token and create commitments.
+    """
+    auth_header = request.headers.get("Authorization")
+    if not auth_header:
+        return jsonify({"error": "Missing Authorization token"}), 401
+
+    user_email = get_user_from_token(auth_header)
+    if not user_email:
+        return jsonify({"error": "Invalid Authorization token"}), 403
+
+    try:
+        request_data = request.get_json()
+        if not isinstance(request_data, list):
+            return jsonify(
+                {"error": "Invalid request format, expected a list"}), 400
```

Comment:
Use codes defined in HTTP_STATUS_CODES_MAPPING in the application/rest/work_shift.py file, instead of hard-coding response codes.